### PR TITLE
internal/cache/cacher test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implements internal/cache/cacher test case.

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/cache/cacher/cacher_test.go [hunk 1]
```diff
@@ -107,31 +102,42 @@ func TestToType(t *testing.T) {
...
+			},
+		},
+		{
+			name: "return Unknown type when the string is invalid",
+			args: args{
+				str: "invalid",
+			},
+			want: want{
+				want: Unknown,
+			},
```
Comment (hlts2) [inline internal/cache/cacher/cacher_test.go:139]: Could you please add test case for empty string?

File: internal/cache/cacher/cacher_test.go [hunk 2]
```diff
@@ -43,25 +43,20 @@ func TestType_String(t *testing.T) {
...
+				want: "gache",
+			},
+		},
+		{
+			name: "return `unknown` when type is unknown",
+			m:    Unknown,
+			want: want{
+				want: "unknown",
+			},
+		},
```
Comment (hlts2) [inline internal/cache/cacher/cacher_test.go:59]: Could you please add test case for invalid m value.
eg.)  m := Type(1000)

File: internal/cache/cacher/cacher.go [hunk 3]
```diff
@@ -34,6 +36,7 @@ func (m Type) String() string {
 	return "unknown"
 }
 
+// ToType returns the type base on the string.
```
Comment (vankichi) [inline internal/cache/cacher/cacher.go:39]:
```suggestion
// ToType returns the type based on the string.
```
